mmlearn/tasks/zero_shot_classification.py

```diff
@@ -0,0 +1,205 @@
+"""Zero-shot and linear probing classification evaluation task."""
+
+import inspect
+import random
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import torch
+import torchmetrics
+from hydra_zen import store
+from pytorch_lightning import LightningModule, Trainer
+from torch.nn import functional
+from torchmetrics import Metric, MetricCollection
+
+from mmlearn.constants import MEDICAL_TEMPLATES
+from mmlearn.datasets.core import Modalities
+from mmlearn.datasets.core.modalities import Modality
+from mmlearn.tasks.hooks import EvaluationHooks
+
+
+@dataclass
+class ClassificationTaskSpec:
+    """Specification for a classification task."""
+
+    metric_name: str
+    query_modality: str
+    top_k: List[int]
+
+
+@store(group="eval_task", provider="mmlearn")
+class ZeroShotClassification(EvaluationHooks):
+    """
+    Zero-shot classification evaluation task.
+
+    This task evaluates the zero-shot classification performance.
+
+    Parameters
+    ----------
+    task_specs : List[ClassificationTaskSpec]
+        A list of classification task specifications.
+    """
+
+    def __init__(
+        self,
+        task_specs: List[ClassificationTaskSpec],
```

Comment:
I think this would be easier to add to the config if it were `Dict[str, ClassificationTaskSpec]`. You will likely not need `configs/experiment/zeroshot.yaml` because you can easily specify it in the command line. It's harder to add new list configs from the command line.